### DISCUSSION TITLE
Keep warm mode

### DISCUSSION THREAD

**Discussion by ryanmurray** [nebhead/PiFire#84] [Ideas]
One thing new traegers have and would be neat if both pifire and James android app had was rather than shutdown option have a keep warm option that at end of internal meat temp it just kicks to a keep warm function

**Comment by nebhead** [comment]
Yep, I think this is definitely something that can be done.  Do you happen to know what traeger does?  Do they just go back to smoke, or back to a specific hold temp?

**Comment by faisalhusain007** [comment]
On my GMG it drops down to 150f and holds it there. Although if left too long depending on what your cooking it can dry it out.

**Reply by ryanmurray** [reply]
I do agree you shouldn't just leave it there for hours but for instance you get an alert on your phone it can keep warm for half hour or so. Probably makes more sense to use it with wrapped meat such as pork shoulder etc

**Comment by ryanmurray** [comment]
Holds at 165

**Reply by nebhead** [reply]
OK, sounds good.  I think the keep warm temp should be adjustable, so, maybe another setting in settings page.  I am working on an installation wizard to make it more user friendly, but will try to get to this after I get that implemented.

**Comment by ryanmurray** [comment]
Awesome by no means is this a rush more just another tool in the belt I
thought we could have :)

**Comment by weberbox** [comment]
I think it sounds like a cool idea. Not sure how much I would use it myself but I always love more options. 

@nebhead
Not sure if you put any thought into this one yet but I submitted a PR to add a keep warm function. It pretty much works the same way as shutdown but it switches the grill to hold mode at the specified temp set in settings. There is also an option to use smoke plus or not in the keep warm mode that can also be changed in settings. I figured just switching to hold mode was much easier then adding a new mode that would work pretty much the same as hold mode anyway. 
This has only been tested on the dev server and not actual hardware yet but I suspect it would be fine. Not sure how long it will take the actual grill to get down to the keep warm temp but I am pretty sure the same would apply to Traeger etc. anyway. 

The PR also includes adding "Manual" mode to the dash.js so all control buttons are hidden when the mode is activated. 

If you had a different idea let me know. I can rework the pr

**Reply by nebhead** [reply]
Outstanding!  I just reviewed and it looks great to me, thanks for implementing this so quickly!
